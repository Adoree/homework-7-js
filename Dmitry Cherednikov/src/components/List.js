import React from 'react';
import PropTypes from 'prop-types';
import PokeItem from '../containers/PokeItem';

const List = ({
  list,
  catched,
  isFetchedAll,
  onClick,
  isFetching,
}) => (
  <div>
    <ul className="list">
      {list.map(elem => (
        <PokeItem catched={catched} key={elem.id} {...elem} />
      ))}
    </ul>
    {!isFetchedAll && !!list.length &&
      <button
        onClick={onClick}
        className="button button-load"
      >
        {isFetching ? 'loading': 'load'}
      </button>
    }
  </div>
);

List.propTypes = {
  isFetchedAll: PropTypes.bool.isRequired,
  isFetching: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  list: PropTypes.array.isRequired,
  catched: PropTypes.bool,
};

export default List;