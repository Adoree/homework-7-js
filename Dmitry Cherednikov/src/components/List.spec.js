import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from './List';

const setup = () => {
  const props = {
    onClick: jest.fn(),
    list: [{
      name: 'bulbasaur',
      id: 1,
      date: '16.07.2018, 15:39:37',
    }, {
      name: 'ivysaur',
      id: 2,
      date: '18.07.2018, 17:46:24',
  }],
    isFetching: false,
    isFetchedAll: false,
  };

  const output = shallow((
    <List { ...props } />
  ));

  return {
    output,
    props,
  }
};

describe('<List />', () => {
  it('renders component', () => {
    const { output } = setup();

    expect(toJson(output)).toMatchSnapshot();
  });

  it('renders button with proper text if all data is not fetched', () => {
    const { output } = setup();

    expect(output.find('button')).toHaveLength(1);
    expect(output.find('button').text()).toBe('load');
  });

  it('button has different text if data is getting fetched', () => {
    const { output } = setup();
    output.setProps({ isFetching: true });

    expect(output.find('button').text()).toBe('loading...');
  });

  it('doesnt render button if all data is fetched', () => {
    const { output } = setup();
    output.setProps({ isFetchedAll: true });

    expect(output.find('button')).toHaveLength(0);
  });

  it('successfully calls the onClick handler', () => {
    const { output, props } = setup();

    output.find('button').simulate('click');
    expect(props.onClick.mock.calls.length).toBe(1);
  })
});