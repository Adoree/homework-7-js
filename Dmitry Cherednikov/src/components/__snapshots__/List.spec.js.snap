// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders component 1`] = `
<div>
  <ul
    className="list"
  >
    <Connect(Component)
      date="16.07.2018, 15:39:37"
      id={1}
      key="1"
      name="bulbasaur"
    />
    <Connect(Component)
      date="18.07.2018, 17:46:24"
      id={2}
      key="2"
      name="ivysaur"
    />
  </ul>
  <button
    className="button button-load"
    disabled={false}
    onClick={[Function]}
  >
    load
  </button>
</div>
`;
